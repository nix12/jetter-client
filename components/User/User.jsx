import React from 'react';
import Can from '../Permission/Can';

const user = props => {
  const { id, username } = props;

  return (
    <Can I="read" this="User">
      <h1>{id}</h1>
      <h1>{`${username}'s Page`}</h1>
    </Can>
  );
};

export default user;
