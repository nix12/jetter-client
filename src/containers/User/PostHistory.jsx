import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import Cookies from 'universal-cookie';

import Can from '../../components/Permissions/Can';
import Post from '../../components/UI/Post/Post';
import Comment from '../../components/UI/Comment/Comment';

import axios from '../../services/axios/axios-forum';
import redirectTo from '../../shared/redirectTo';

const PostHistory = props => {
  const { value, index } = props;

  const [data, setData] = useState([]);
  const [updatePost, setUpdatePost] = useState(false);
  const [updateComment, setUpdateComment] = useState(false);

  const userId = useSelector(state => state.auth.currentUser.userId);
  const username = useSelector(state => state.auth.currentUser.username);

  useEffect(() => {
    const fetchPostHistory = async () => {
      const postHistory = await axios.get(
        `/api/voters/${username}/post_history`
      );

      setData(postHistory.data.history);
    };

    if (username) {
      fetchPostHistory();
    }
  }, [username, updateComment, updatePost]);

  return (
    value === index && (
      <Can I="read" this={{ __type: 'User', id: userId }}>
        <h1>Post History</h1>
        {data.map(post => {
          if (!post.title) {
            return (
              <Comment
                key={post.hash_id}
                comment={{ __type: 'Comment', ...post }}
                body={post.body}
                comments={post.comments_count}
                createdAt={post.created_at}
                commentId={post.hash_id}
                jetId={post.jet_id}
                updatedAt={post.updated_at}
                username={post.voter_id}
                score={post.cached_votes_score}
                depth={post.depth}
                setUpdateComment={setUpdateComment}
                style={{ backgroundColor: 'yellow' }}
              />
            );
          }

          return (
            <Post
              key={post.hash_id}
              post={{
                __type: post.type === 'Post' ? 'Link' : 'Text',
                ...post
              }}
              type={post.type.toLowerCase()}
              body={post.body}
              comments={post.comments_count}
              createdAt={post.created_at}
              jetId={post.jet_id}
              postId={post.hash_id}
              title={post.title}
              uri={post.uri}
              updatedAt={post.updated_at}
              username={post.voter_id}
              score={post.cached_votes_score}
              setUpdatePost={setUpdatePost}
              style={{ backgroundColor: 'lightblue' }}
            />
          );
        })}
      </Can>
    )
  );
};

PostHistory.getInitialProps = async ({ res, isServer }) => {
  const cookies = new Cookies();
  const token = cookies.get('token');

  if (!token && !isServer) {
    redirectTo('/login');
  }

  return {};
};

export default PostHistory;
