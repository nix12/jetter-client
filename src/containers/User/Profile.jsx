import React from 'react';
import { useSelector } from 'react-redux';

import Can from '../../components/Permissions/Can';
import { ability } from '../../services/casl/ability';

const Profile = props => {
  const { value, index } = props;
  const user = useSelector(state => state.auth.currentUser);
  const username = useSelector(state => state.auth.currentUser.username);
  const userId = useSelector(state => state.auth.currentUser.userId);
  const roles = useSelector(state => state.auth.currentUser.roles);

  const rolesList = roles
    ? roles.map((role, index) => (
        <span key={index} style={{ display: 'block' }}>
          role {role}
        </span>
      ))
    : null;

  const abilities = ability.rules.map((rule, index) => (
    <span key={index} style={{ display: 'block' }}>
      rules [{rule.actions}: {rule.subject}]
    </span>
  ));

  return (
    value === index && (
      <div>
        <Can I="read" this={{ __type: 'User', id: user.userId }}>
          <h1>My Profile</h1>
          <h1>{user.userId}</h1>
          <h1>{`${user.username}'s Page`}</h1>
          <div>
            <span style={{ display: 'block' }}>ID: {userId}</span>
            <span style={{ display: 'block' }}>Username: {username}</span>
            <span style={{ display: 'block' }}>{rolesList}</span>
            {abilities}
          </div>
        </Can>
      </div>
    )
  );
};

export default Profile;
