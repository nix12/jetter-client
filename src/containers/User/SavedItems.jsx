import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import Cookies from 'universal-cookie';

import Can from '../../components/Permissions/Can';
import Post from '../../components/UI/Post/Post';
import Comment from '../../components/UI/Comment/Comment';

import axios from '../../services/axios/axios-forum';
import redirectTo from '../../shared/redirectTo';

const SavedItems = props => {
  const { value, index } = props;
  // const user = useSelector(state => state.auth.currentUser);

  const [data, setData] = useState([]);
  const [updatePost, setUpdatePost] = useState(false);
  const [updateComment, setUpdateComment] = useState(false);

  const userId = useSelector(state => state.auth.currentUser.userId);
  const username = useSelector(state => state.auth.currentUser.username);

  useEffect(() => {
    const fetchSavedItems = async () => {
      const savedItems = await axios.get(`/api/voters/${username}/saved_items`);

      setData(savedItems.data.saved_items);
    };

    if (username) {
      fetchSavedItems();
    }
  }, [username, updateComment, updatePost]);

  return (
    value === index && (
      <Can I="read" this={{ __type: 'User', id: userId }}>
        <h1>Saved Items</h1>
        {data.map(post => {
          if (!post.title) {
            return (
              <Comment
                key={post.hash_id}
                comment={{ __type: 'Comment', ...post }}
                body={post.body}
                comments={post.comments_count}
                createdAt={post.created_at}
                commentId={post.hash_id}
                jetId={post.jet_id}
                updatedAt={post.updated_at}
                username={post.voter_id}
                score={post.cached_votes_score}
                depth={post.depth}
                setUpdateComment={setUpdateComment}
                style={{ backgroundColor: 'yellow' }}
              />
            );
          }

          return (
            <Post
              key={post.hash_id}
              post={{
                __type: post.type === 'Post' ? 'Link' : 'Text',
                ...post
              }}
              type={post.type ? post.type.toLowerCase() : null}
              body={post.body}
              comments={post.comments_count}
              createdAt={post.created_at}
              jetId={post.jet_id}
              postId={post.hash_id}
              title={post.title}
              uri={post.uri}
              updatedAt={post.updated_at}
              username={post.voter_id}
              score={post.cached_votes_score}
              setUpdatePost={setUpdatePost}
              style={{ backgroundColor: 'lightblue' }}
            />
          );
        })}
      </Can>
    )
  );
};

SavedItems.getInitialProps = async ({ res, isServer }) => {
  const cookies = new Cookies();
  const token = cookies.get('token');

  if (!token && !isServer) {
    redirectTo('/login');
  }

  return {};
};

export default SavedItems;
