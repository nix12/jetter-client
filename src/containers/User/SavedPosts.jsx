import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import Cookies from 'universal-cookie';

import Can from '../../components/Permissions/Can';

import axios from '../../services/axios/axios-forum';
import redirectTo from '../../shared/redirectTo';

const SavedPosts = props => {
  const { value, index } = props;
  const user = useSelector(state => state.auth.currentUser);

  const [data, setData] = useState([]);
  const [updatePost, setUpdatePost] = useState(false);
  const [updateComment, setUpdateComment] = useState(false);

  const userId = useSelector(state => state.auth.currentUser.userId);
  const username = useSelector(state => state.auth.currentUser.username);

  useEffect(() => {
    const fetchPostHistory = async () => {
      const postHistory = await axios.get(
        `/api/voters/${username}/saved_posts`
      );

      setData(postHistory.data.history);
    };

    if (username) {
      fetchPostHistory();
    }
  }, [username, updateComment, updatePost]);

  return (
    value === index && (
      <div>
        <Can I="read" this={{ __type: 'User', id: user.userId }}>
          <h1>Saved Posts</h1>
        </Can>
      </div>
    )
  );
};

SavedPosts.getInitialProps = async ({ res, isServer }) => {
  const cookies = new Cookies();
  const token = cookies.get('token');

  if (!token && !isServer) {
    redirectTo('/login');
  }

  return {};
};

export default SavedPosts;
