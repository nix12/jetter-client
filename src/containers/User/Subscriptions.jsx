import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';

import Jet from '../../components/UI/Jet/Jet';

import { listSubscriptions } from '../../store/actions/index';

const Subscriptions = props => {
  const { value, index } = props;

  const [subscriptions, setSubscriptions] = useState([]);

  const dispatch = useDispatch();

  useEffect(() => {
    const fetchSubscriptions = async () => {
      await dispatch(listSubscriptions()).then(response => {
        console.log('response', response);
        setSubscriptions(response);
      });
    };

    fetchSubscriptions();
  }, []);

  return (
    value === index && (
      <div>
        <h1>Subscriptions</h1>
        {console.log('asdf', subscriptions)}
        {subscriptions &&
          subscriptions.map(subscription => (
            <Jet
              key={subscription.created_at}
              jet={{ __type: 'Jet', ...subscription }}
              jetId={subscription.jet_id}
              createdAt={subscription.created_at}
            />
          ))}
      </div>
    )
  );
};

export default Subscriptions;
