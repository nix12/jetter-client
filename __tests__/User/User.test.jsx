import React from 'react';
import { shallow } from 'enzyme';
import User from '../../components/User/User';

describe('user', () => {
  it('should render user component', () => {
    expect(shallow(<User />)).toMatchSnapshot();
  });
});
