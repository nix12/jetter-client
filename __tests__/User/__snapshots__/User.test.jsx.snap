// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user should render user component 1`] = `ShallowWrapper {}`;
